### DISCUSSION TITLE
Vulnerability update is not auto-merged

### DISCUSSION THREAD

**Discussion by northern-mindset** [renovatebot/renovate#10221] [Request Help]
Renovate config (abridged):

```
{
  "extends": [
    ":automergeMinor",
    ":dependencyDashboard",
    ":enableVulnerabilityAlertsWithLabel(security)"
  ],
  "schedule": ["on the 20th day of the month"],
  "vulnerabilityAlerts": {
    "schedule": ["at any time"]
  },
  "major": {
    "dependencyDashboardApproval": true
  }
}
```

I got a pull request for a vulnerability alert, a patch update: https://github.com/PureBhaktiArchive/audioseva/pull/663

ws | 7.4.1&nbsp;-&gt;&nbsp;7.4.6
-- | --

It says:
>Automerge: Disabled by config.

From [Renovate logs](https://app.renovatebot.com/dashboard#github/PureBhaktiArchive/audioseva/364944147): 

`DEBUG: PR is not configured for automerge(branch="renovate/npm-ws-vulnerability")`

Why? My config is supposed to allow automerging for minor and patch updates.

**Comment by rarkins** [comment]
Could be a bug in the automergeMinor preset. That update is a patch, but should be automerged too.

**Reply by HonkingGoose** [reply]
Could it be that their `schedule` is limiting Renovate auto-merge to a certain day only?

```json5
  "schedule": ["on the 20th day of the month"],
```

---

**EDIT:**

Whoops, they seem to allow automerge for security updates at all times though:

```json5
  "vulnerabilityAlerts": {
    "schedule": ["at any time"]
  },
```

---

**EDIT to the EDIT:**

The `automergeMinor` presets seems OK to my eyes:

https://github.com/renovatebot/renovate/blob/ba23d174eb301e138989a8a395134fe713729b06/lib/config/presets/internal/default.ts#L279-L293

We do have a migration, maybe that's doing something wrong?

https://github.com/renovatebot/renovate/blob/ba23d174eb301e138989a8a395134fe713729b06/lib/config/migration.ts#L302-L306

**Reply by northern-mindset** [reply]
Maybe `vulnerabilityAlerts` does not inherit global `automerge` setting and needs explicit setting like `lockFileMaintenance`?

**Reply by rarkins** [reply]
Can you create a [minimal reproduction repository](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) for this?

**Reply by northern-mindset** [reply]
Maybe after some time. However [my guess above](https://github.com/renovatebot/renovate/discussions/10221#discussioncomment-800981) seems to be confirmed: when I [explicitly allowed automerge for vulnerability alerts](https://github.com/PureBhaktiArchive/audioseva/commit/fbc09482117cfb208572d482502be283f2ac55e4), [the same PR](https://github.com/PureBhaktiArchive/audioseva/pull/663) was turned into “Automerge: Enabled”.
